core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -239,3 +235,77 @@ private fun <ReqT : ResourceRequirement> detectRequirementConflicts(
     }
     return conflictGroups
 }
+
+enum class EventType { BEGIN, END }
+class Event (eventType: EventType, time : Double){
+
+    val time: Double = time
+    val eventType : EventType = eventType
+}
+
+fun mergeMap(resources : HashMap<Set<Long>, MutableList<Conflict>>, conflictType : ConflictType): MutableList<Conflict> {
+    // sort and merge conflicts with overlapping time ranges
+    val newConflicts = mutableListOf<Conflict>()
+    for ((trainIds, conflicts) in resources)
+    {
+        // create an event list and sort it
+        val events = mutableListOf<Event>()
+        for (conflict in conflicts)
+        {
+            events.add(Event(EventType.BEGIN, conflict.startTime))
+            events.add(Event(EventType.END, conflict.endTime))
+        }
+        events.sortWith(Comparator<Event> {a, b ->
+            when { a.time > b.time -> 1
+                a.time < b.time -> -1
+                else -> 0}
+        })
+
+        var eventCount = 0
+        var eventBeginning = 0.0
+        for (event in events)
+        {
```

Comment:
The [standard coding style](https://kotlinlang.org/docs/coding-conventions.html#file-annotations) for kotlin has the opening brackets on the same line as conditions, just like in java